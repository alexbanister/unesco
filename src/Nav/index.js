/* eslint no-unused-vars: 0 */

import React from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';
import { withRouter, Link } from 'react-router-dom';

const Header = () => {
  return (
    <nav className='header__nav'>
      <ul className='header__nav__list'>
        <li className='header__nav__list__item'>
          <Link to='/dashboard'>dashboard</Link>
        </li>
        <li className='header__nav__list'>
          <Link to='/favorites'>favorites</Link>
        </li>
        <li className='header__nav__list'>
          <Link to='/visited'>visited</Link>
        </li>
        <li className='header__nav__list'>
          <Link to='/wants'>want to go</Link>
        </li>
        <li className='header__nav__list'>
          <Link to='/profile'>profile</Link>
        </li>
      </ul>
    </nav>
  );
};

Header.propTypes = {};

export default Header;
