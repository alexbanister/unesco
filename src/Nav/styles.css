.header__nav__list {
  list-style: none; }

.header__nav__list__item {
  text-decoration: none; }
