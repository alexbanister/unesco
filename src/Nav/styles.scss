@import "../variables.scss";

.nav {
  width: 100%;
  height: 100%;
}

.nav__list {
  height: 100%;
  width: 60%;
  margin-left: 20%;
  list-style: none;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.nav__list__item {
  height: 100%;
  flex-grow: 1;
  white-space: nowrap;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
}

.nav__link:hover{
  color: $nav-link-highlight;
}

.nav__link {
  height: 100%;
  text-transform: uppercase;
  text-decoration: none;
  display: flex;
  align-items: center;
  justify-content: center;
  color: $nav-link;
}

.nav-link--selected {
  color: $nav-link-highlight;
  border-bottom: 4px solid $nav-link-highlight;
  margin-bottom: -4px;
}