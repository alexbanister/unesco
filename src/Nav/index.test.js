import { shallow } from 'enzyme';
import React from 'react';
import Nav from './index';


describe('Nav', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<Nav />);

    expect(wrapper).toMatchSnapshot();
  });
});