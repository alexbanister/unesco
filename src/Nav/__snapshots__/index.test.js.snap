// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="nav__list"
>
        <li
                className="nav__list__item"
        >
                <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link"
                        to="/dashboard"
                >
                        dashboard
                </NavLink>
        </li>
        <li
                className="nav__list__item"
        >
                <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link"
                        to="/favorites"
                >
                        favorites
                </NavLink>
        </li>
        <li
                className="nav__list__item"
        >
                <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link"
                        to="/visited"
                >
                        visited
                </NavLink>
        </li>
        <li
                className="nav__list__item"
        >
                <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link wants"
                        to="/wants"
                >
                        want to go
                </NavLink>
        </li>
        <li
                className="nav__list__item"
        >
                <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link profile"
                        to="/profile"
                >
                        profile
                </NavLink>
        </li>
</ul>,
      "className": "nav",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav__list__item"
>
            <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link"
                        to="/dashboard"
            >
                        dashboard
            </NavLink>
</li>,
          <li
            className="nav__list__item"
>
            <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link"
                        to="/favorites"
            >
                        favorites
            </NavLink>
</li>,
          <li
            className="nav__list__item"
>
            <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link"
                        to="/visited"
            >
                        visited
            </NavLink>
</li>,
          <li
            className="nav__list__item"
>
            <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link wants"
                        to="/wants"
            >
                        want to go
            </NavLink>
</li>,
          <li
            className="nav__list__item"
>
            <NavLink
                        activeClassName="nav-link--selected"
                        ariaCurrent="true"
                        className="nav__link profile"
                        to="/profile"
            >
                        profile
            </NavLink>
</li>,
        ],
        "className": "nav__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="nav-link--selected"
              ariaCurrent="true"
              className="nav__link"
              to="/dashboard"
>
              dashboard
</NavLink>,
            "className": "nav__list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav-link--selected",
              "ariaCurrent": "true",
              "children": "dashboard",
              "className": "nav__link",
              "to": "/dashboard",
            },
            "ref": null,
            "rendered": "dashboard",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="nav-link--selected"
              ariaCurrent="true"
              className="nav__link"
              to="/favorites"
>
              favorites
</NavLink>,
            "className": "nav__list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav-link--selected",
              "ariaCurrent": "true",
              "children": "favorites",
              "className": "nav__link",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "favorites",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="nav-link--selected"
              ariaCurrent="true"
              className="nav__link"
              to="/visited"
>
              visited
</NavLink>,
            "className": "nav__list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav-link--selected",
              "ariaCurrent": "true",
              "children": "visited",
              "className": "nav__link",
              "to": "/visited",
            },
            "ref": null,
            "rendered": "visited",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="nav-link--selected"
              ariaCurrent="true"
              className="nav__link wants"
              to="/wants"
>
              want to go
</NavLink>,
            "className": "nav__list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav-link--selected",
              "ariaCurrent": "true",
              "children": "want to go",
              "className": "nav__link wants",
              "to": "/wants",
            },
            "ref": null,
            "rendered": "want to go",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="nav-link--selected"
              ariaCurrent="true"
              className="nav__link profile"
              to="/profile"
>
              profile
</NavLink>,
            "className": "nav__list__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "nav-link--selected",
              "ariaCurrent": "true",
              "children": "profile",
              "className": "nav__link profile",
              "to": "/profile",
            },
            "ref": null,
            "rendered": "profile",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav__list"
>
          <li
                    className="nav__list__item"
          >
                    <NavLink
                              activeClassName="nav-link--selected"
                              ariaCurrent="true"
                              className="nav__link"
                              to="/dashboard"
                    >
                              dashboard
                    </NavLink>
          </li>
          <li
                    className="nav__list__item"
          >
                    <NavLink
                              activeClassName="nav-link--selected"
                              ariaCurrent="true"
                              className="nav__link"
                              to="/favorites"
                    >
                              favorites
                    </NavLink>
          </li>
          <li
                    className="nav__list__item"
          >
                    <NavLink
                              activeClassName="nav-link--selected"
                              ariaCurrent="true"
                              className="nav__link"
                              to="/visited"
                    >
                              visited
                    </NavLink>
          </li>
          <li
                    className="nav__list__item"
          >
                    <NavLink
                              activeClassName="nav-link--selected"
                              ariaCurrent="true"
                              className="nav__link wants"
                              to="/wants"
                    >
                              want to go
                    </NavLink>
          </li>
          <li
                    className="nav__list__item"
          >
                    <NavLink
                              activeClassName="nav-link--selected"
                              ariaCurrent="true"
                              className="nav__link profile"
                              to="/profile"
                    >
                              profile
                    </NavLink>
          </li>
</ul>,
        "className": "nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="nav__list__item"
>
              <NavLink
                            activeClassName="nav-link--selected"
                            ariaCurrent="true"
                            className="nav__link"
                            to="/dashboard"
              >
                            dashboard
              </NavLink>
</li>,
            <li
              className="nav__list__item"
>
              <NavLink
                            activeClassName="nav-link--selected"
                            ariaCurrent="true"
                            className="nav__link"
                            to="/favorites"
              >
                            favorites
              </NavLink>
</li>,
            <li
              className="nav__list__item"
>
              <NavLink
                            activeClassName="nav-link--selected"
                            ariaCurrent="true"
                            className="nav__link"
                            to="/visited"
              >
                            visited
              </NavLink>
</li>,
            <li
              className="nav__list__item"
>
              <NavLink
                            activeClassName="nav-link--selected"
                            ariaCurrent="true"
                            className="nav__link wants"
                            to="/wants"
              >
                            want to go
              </NavLink>
</li>,
            <li
              className="nav__list__item"
>
              <NavLink
                            activeClassName="nav-link--selected"
                            ariaCurrent="true"
                            className="nav__link profile"
                            to="/profile"
              >
                            profile
              </NavLink>
</li>,
          ],
          "className": "nav__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="nav-link--selected"
                ariaCurrent="true"
                className="nav__link"
                to="/dashboard"
>
                dashboard
</NavLink>,
              "className": "nav__list__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "nav-link--selected",
                "ariaCurrent": "true",
                "children": "dashboard",
                "className": "nav__link",
                "to": "/dashboard",
              },
              "ref": null,
              "rendered": "dashboard",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="nav-link--selected"
                ariaCurrent="true"
                className="nav__link"
                to="/favorites"
>
                favorites
</NavLink>,
              "className": "nav__list__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "nav-link--selected",
                "ariaCurrent": "true",
                "children": "favorites",
                "className": "nav__link",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "favorites",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="nav-link--selected"
                ariaCurrent="true"
                className="nav__link"
                to="/visited"
>
                visited
</NavLink>,
              "className": "nav__list__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "nav-link--selected",
                "ariaCurrent": "true",
                "children": "visited",
                "className": "nav__link",
                "to": "/visited",
              },
              "ref": null,
              "rendered": "visited",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="nav-link--selected"
                ariaCurrent="true"
                className="nav__link wants"
                to="/wants"
>
                want to go
</NavLink>,
              "className": "nav__list__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "nav-link--selected",
                "ariaCurrent": "true",
                "children": "want to go",
                "className": "nav__link wants",
                "to": "/wants",
              },
              "ref": null,
              "rendered": "want to go",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="nav-link--selected"
                ariaCurrent="true"
                className="nav__link profile"
                to="/profile"
>
                profile
</NavLink>,
              "className": "nav__list__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "nav-link--selected",
                "ariaCurrent": "true",
                "children": "profile",
                "className": "nav__link profile",
                "to": "/profile",
              },
              "ref": null,
              "rendered": "profile",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
